### PR TITLE
feat(medusa-plugin-mailchimp): Migrate to TS + add update endpoint

### PR DESCRIPTION
**Pull request by tomekpur**
As proposed in [Addition of endpoint to mailchimp plugin and refactor to typescript #3985](https://github.com/medusajs/medusa/discussions/3985
)
- Refactored the .js files to .ts files
- Added typing
- Used 'WrapHandler' util instead of specific middleware in plugin
- Added additional PUT method to the /mailchimp/subscribe endpoint.
- Updated documentation where relevant.

### CODE REVIEW

File: packages/medusa-plugin-mailchimp/src/services/mailchimp.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+   * Updates an email to a newsletter.
+   * @param {string} email - email to use for the subscription
+   * @param {Object} data - additional data (see https://mailchimp.com/developer/marketing/api/list-merges/)
+   * @return {Promise} result of newsletter subscription
+   */
+
+  async subscribeNewsletterUpdate(
+    email: string,
+    data: any,
+    status_if_new?: string,
```
Comment (olivermrbl) [inline packages/medusa-plugin-mailchimp/src/services/mailchimp.ts]: nit: We use camelCasing for method arguments
```suggestion
    statusIfNew?: string,
```

**Review by olivermrbl — COMMENTED** [review]
This is great - appreciate the contribution! ❤️

**Review by olivermrbl — COMMENTED** [review]
This is great - appreciate the contribution! ❤️

**Comment by tomekpur** [comment]
@olivermrbl Cool, thanks and my pleasure. I'll update the yarn.lock for the CI and process your comment after the weekend 👍🏻

**Review by olivermrbl — APPROVED** [review]
LGTM - appreciate the contribution @tomekpur <3

**Comment by tomekpur** [comment]
@olivermrbl Is the error in the integration tests something I should/can fix? It looks like something that isn't caused by any of my changes?

**Comment by olivermrbl** [comment]
@tomekpur Will try to give the CI another spin. Can you clean up the conflicts in the mean time? :)

**Comment by tomekpur** [comment]
@olivermrbl Done.

**Pull request merged**